core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BillingProfileManagerDAO.scala

```diff
@@ -125,24 +119,60 @@ class BillingProfileManagerDAOImpl(samDAO: SamDAO,
       )
       .flatMap {
         case true =>
-          Future.successful(
-            Seq(
-              RawlsBillingProject(
-                RawlsBillingProjectName(azureConfig.billingProjectName),
-                CreationStatuses.Ready,
-                None,
-                None,
-                azureManagedAppCoordinates = Some(
-                  AzureManagedAppCoordinates(
-                    UUID.fromString(azureConfig.azureTenantId),
-                    UUID.fromString(azureConfig.azureSubscriptionId),
-                    azureConfig.azureResourceGroupId
+          val profileApi = apiClientProvider.getProfileApi(ctx)
+          Future.successful(profileApi.listProfiles(0, 1000).getItems.asScala.toSeq)
+        case _ => Future.successful(Seq())
+      }
+  }
+
+  def getHardcodedAzureBillingProject(samUserResources: Seq[SamUserResource], userInfo: UserInfo)(implicit
+    ec: ExecutionContext
+  ): Future[Seq[RawlsBillingProject]] = {
+    if (!config.multiCloudWorkspacesEnabled) {
+      return Future.successful(Seq())
+    }
+
+    val azureConfig = config.azureConfig match {
+      case None =>
+        logger.warn("Multicloud workspaces enabled but no azure config setup, returning empty list of billing profiles")
+        return Future.successful(Seq())
+      case Some(value) => value
+    }
+
+    for {
+      billingProjects <- samDAO
+        .userHasAction(
+          SamResourceTypeNames.managedGroup,
+          azureConfig.alphaFeatureGroup,
+          SamResourceAction("use"),
+          userInfo
+        )
+        .flatMap {
+          case true =>
+            // Will remove after users can create Azure-backed Billing Accounts via Terra.
+            Future.successful(
+              Seq(
+                RawlsBillingProject(
+                  RawlsBillingProjectName(azureConfig.billingProjectName),
+                  CreationStatuses.Ready,
+                  None,
+                  None,
+                  azureManagedAppCoordinates = Some(
+                    AzureManagedAppCoordinates(
+                      UUID.fromString(azureConfig.azureTenantId),
+                      UUID.fromString(azureConfig.azureSubscriptionId),
+                      azureConfig.azureResourceGroupId
+                    )
                   )
                 )
               )
             )
-          )
-        case false =>
-          Future.successful(Seq.empty)
-      }
+          case false =>
+            Future.successful(Seq.empty)
+        }
+    } yield billingProjects.filter { bp =>
+      samUserResources.map(_.resourceId).contains(bp.projectName.value)
```

Comment:
This filtering was in the original listBillingProfiles method, but I believe it is only needed for the hardcoded account because of the code at https://github.com/broadinstitute/rawls/pull/1994/files#diff-727d553ef55a7aabb5b0d0bca2e6a6fe44422feef4c7ad5ebb9c86bcd92b6572L179.

It also seems good that with the exception of this temporary hack method, we are no longer passing Sam resources into the DAO methods.